starfish/core/util/logging.py

```diff
@@ -1,12 +1,44 @@
 import platform
 from functools import lru_cache
 from json import JSONEncoder
-from typing import Mapping
+from typing import List, Mapping
 
 import pkg_resources
 
 import starfish.core
-from starfish.core.types import CORE_DEPENDENCIES
+from starfish.core.types import CORE_DEPENDENCIES, LOG
+
+
+class Log:
```

Comment:
Love the idea. Is this only used by spot finding? My intuition is that this should trigger a bigger refactor.

Reply:
Nah this is used by everything. This PR was basically the "bigger" logging refactor needed to keep everything working throughout the refactor